File: src/server/Network/SendRpc.js
import PacketIds from "Network/PacketIds";
import * as msgpack from "@msgpack/msgpack";

function SendRpc(
  socket,
  name,
  response
) {
  if (socket && socket.readyState !== socket.OPEN) return;

  socket.send(msgpack.encode({
    opcode: PacketIds.PACKET_RPC,
    name,
    response
  }));
}

export default SendRpc;

File: src/v2/pyZOMBIA.io/src/server/Network/SendRpc.py
from Network.PacketIds import packets

import msgpack
import asyncio
import websockets

def SendRpc(socket, packet):
  try:
    asyncio.get_event_loop().run_until_complete(socket.send(msgpack.packb({
      "opcode": packets["PACKET_RPC"],
      "response": packet
    })))
  except websockets.exceptions.ConnectionClosed:
    try:
      del data["clients"][Util.socketUid(websocket)]
    except Exception:
      pass
